console/frontend/src/main/frontend/src/app/views/storage/storage-list/storage-list-dt/storage-list-dt.component.html

```diff
@@ -5,10 +5,33 @@
       <a [routerLink]="['messages', message.id]" class="btn btn-info btn-xs" type="button"
         ><i class="fa fa-file-text-o"></i> View</a
       >
+      <button
+        appHasAccessToLink="downloadMessage"
+        title="Download Message"
+        (click)="storageService.downloadMessage(message.id)"
+        class="btn btn-info btn-xs"
+        type="button"
+      >
+        <i class="fa fa-arrow-circle-o-down"></i> Download
+      </button>
+      @if (storageService.storageParams.processState === 'InProcess') {
+        <button
+          appHasAccessToLink="changeMessagesProcessState"
+          [ladda]="message.processing"
+          data-style="slide-down"
+          title="Move Message to Error"
+          (click)="moveMessage()"
```

Comment:
Moet hier niet een message.id in?

Reply:
Nee, deze component wordt voor elke row in de datatable aangeroepen en houd zelf al een reference naar het message object (inclusief message id), de function heeft het niet extra als argument nodig :)